data/categories.txt

```diff
@@ -1,4 +1,15 @@
 # Category groups referenced in many categories, should be defined before being referenced.
+# Syntax:
+# @   - string that starts with this symbol indicates the beginning of the group translations, which are used for the spotlight search.
+# :   - symbol that is used to separate the name of the language and translation (or group of translated synonyms). 
+#       Language name is located on the left side of ':', and the translation on the right.
+# |   - symbol that is used to separate several translated synonyms.
+# 1-9 - numbers from one to nine, indicate the weight of the translated word for assumptions in search queries. 
+#       Located immediately after the separating characters ':' or '|'. Acceptable use only one digit for one word.
+# ^   - symbol indicates that the translation should be used in the user interface to identify the category in PP and editor. 
+#       Located either immediately after the separating characters ':', '|' or if there is a number indicating the weight 
```

Comment:
It would be much nicer and better to specify the grammar in the BNF form. I.e.
```
rule => lang ':' translation-list
translation-list => translation-item ('|' translation-list)*
translation-item => ux-identifier? prefix-length? translation
ux-identifier => '^'
prefix-length => [1-9]
translation => ...
```